src/TranslatorInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Dhii\I18n;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use ArrayAccess;
+use Dhii\Data\ValueAwareInterface;
+
+/**
+ * Something that can act as a string translator.
+ *
+ * @since [*next-version*]
+ */
+interface TranslatorInterface
+{
+    /**
+     * Translates a string.
+     *
+     * If values are given, MUST attempt to use them for substituting placeholders in the string.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $string The string to translate.
+     * @param array|ArrayAccess|null $params A map of values to use for placeholder substitution, if any.
+     * @param string|ValueAwareInterface|null $context A context for the string, if any.
+     *
+     * @return string The translated string, possibly with placeholders substituted.
```

Comment:
Maybe you misunderstood me. I meant specifying a return type of `string|Stringable`. It would allow an implementation to return an object that can resolve itself into the translated string. An example would be a translator that can translate blocks.